src/main/kotlin/no/nav/tiltak/tiltaknotifikasjon/persondata/PersondataService.kt

```diff
@@ -0,0 +1,26 @@
+package no.nav.tiltak.tiltaknotifikasjon.persondata
+
+import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenService
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties
+import no.nav.team_tiltak.felles.persondata.PersondataClient
+import no.nav.team_tiltak.felles.persondata.pdl.domene.Diskresjonskode
+import org.springframework.context.annotation.Profile
+import org.springframework.stereotype.Service
+
+@Service
+@Profile("prod-gcp", "dev-gcp", "dockercompose")
+class PersondataService(
+    clientConfigurationProperties: ClientConfigurationProperties,
+    persondataProperties: PersondataProperties,
+    val oAuth2AccessTokenService: OAuth2AccessTokenService,
+) {
+    private val clientProperties = clientConfigurationProperties.registration["pdl-api"]
+
+    private val persondataClient =
+        PersondataClient(persondataProperties.uri) { clientProperties?.let { oAuth2AccessTokenService.getAccessToken(it).access_token } }
+
+    fun hentDiskresjonskode(fnr: String): Diskresjonskode {
+        return persondataClient.hentDiskresjonskode(fnr).orElse(Diskresjonskode.UGRADERT)
```

Comment:
Burde denne kanskje kaste en exception om vi ikke får noe resultat istedenfor å bare anta at det er ugradert?

I Tiltaksgjennomføring bruker vi bare diskresjonskoden for "info"-bokser, så der er det egentlig greit at den bare defaulter til ugradert. Men i prosess så kaster vi en exception slik at vi må prøve på ny dersom vi ikke vet om vi skal sladde eller ikke.

Reply:
Ja, det er et godt poeng. Var litt copy-paste fra @Oddsor sin implementasjon i tiltak-refusjon-api. Men det er vel litt samme usecase der som i tiltaksgjennomføring.